pkg/catalog/catalog.go

```diff
@@ -129,8 +131,67 @@ func getDefaultBuilderImages(client *occlient.Client) ([]CatalogImage, error) {
 		imageStreamTagMap[imageStreamTag.Name] = imageStreamTag
 	}
 
-	var builderImages []CatalogImage
+	builderImages := getBuildersFromImageStreams(imageStreams, imageStreamTagMap)
+
+	return builderImages, nil
+}
 
+// SpliceSupportedTags splits the tags in to fully supported and unsupported tags
+func SpliceSupportedTags(catalogImage CatalogImage) ([]string, []string) {
+
+	var supTag, nonSupTag []string
+	tagMap := createImageMap(catalogImage.imageStreamRef.Spec.Tags)
+	for _, tag := range catalogImage.NonHiddenTags {
+		imageName := tagMap[tag]
```

Comment:
Would it be a better idea to declare `imageName` before the `for` loop instead of creating it every time during each iteration of `for` loop? Just like we've done for `supTag` and `nonSupTag` variables? Honestly, this is a question because I have no clue about best practices in Go.

Reply:
`imageName` is using the `tag` loop variable hence it cannot be declared outside of the for loop @dharmit

Reply:
Okay, I guess there's no value being added by doing it outside the `for` loop.